stdlib/globals.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//import ecrecover from std;
+//import sha256 from std;
+//import chainid from std;
+
+contract C{
+
+function addmod(uint x, uint y, uint k) public returns (uint r) {
+  require(k != 0);
+  assembly {
+    r := addmod(x, y, k)
+  }
+}
+
+function mulmod(uint x, uint y, uint k) public returns (uint r) {
+  require(k != 0);
+  assembly {
+    r := mulmod(x, y, k)
+  }
+}
+
+function keccak256(bytes memory input) public returns (bytes32 ret) {
+  assembly {
+    ret := keccak256(add(input, 32), mload(input))
+  }
+}
+
+function sha256(bytes memory input) public returns (bytes32 ret) {
+  assembly {
+    let success := staticcall(gas(), 2, add(input, 32), mload(input), 0, 32)
+    if iszero(success) { revert(0, 0) }
+    ret := mload(0)
+  }
+}
+
+function ripemd160(bytes memory input) public returns (bytes20 ret) {
+  assembly {
+    let success := staticcall(gas(), 3, add(input, 32), mload(input), 0, 32)
+    if iszero(success) { revert(0, 0) }
+    // TODO: check byteorder
+    ret := mload(0)
+  }
+}
+
+function ecrecover(bytes32 hash, uint8 v, bytes32 r, bytes32 s) public returns (address addr) {
+  assembly {
+    // Prepare input
+    // We are using the free memory pointer.
+    let input := mload(0x40)
```

Comment:
Move this into a function?

Reply:
So far these files were independently working, and the assembly file of course does not work yet. But yes, eventually it should use the assembly helpers.